src/os/bluestore/bluestore_types.h

```diff
@@ -283,7 +283,7 @@ struct bluestore_blob_t {
     }
     b_len += b_off;
     while (b_len) {
-      if (!p->is_valid()) {
+      if (p == extents.end() || !p->is_valid()) {
```

Comment:
@xiexingguo, I'm sorry for waving back and forth but probably assert is still better here and above. We should always have all pextents present in the list - either valid or not. The lack of specific pextent (or out of range access) should be considered as an unexpected case and hence asserted IMHO. Thus desired fix should be just an assert before p->is_valid call. What do you think?